### ISSUE TITLE
Is it possible to plot two scalar summary in one plot?

### ISSUE THREAD

**Issue by shawnLeeZX** [tensorflow/tensorflow#329]
Hi, 

I have spent some time writing scripts to do stat in Caffe and this let me know Tensorboard is so amazing! It is actually one of the major reasons that makes me switch to tensorflow quickly. 

I just wonder could an option be added to plot more than two scalars in the same plot? More specifically. it could allow:

1. training loss and test loss being plotted together
2. regularization loss and top level loss to be plotted together.
3. maybe adding accuracy on the right vertical axis

Just name a few. This would help a lot.

**Comment by girving** [comment]
Cc @danmane.

**Comment by danmane** [comment]
Yeah, we definitely want to make comparing different charts easier. No timeline for when we'll have this, though.

**Comment by zhang8473** [comment]
Yes, very necessary, now I have to use matplotlib to make such plots.

**Comment by danmane** [comment]
As a temporary workaround, you can make the plot in matplotlib, and then inject it into a TensorFlow image_summary so that it will display an updated copy in TensorBoard.

**Comment by zhang8473** [comment]
....why not I directly go and see the plot made in matplotlib?

**Comment by yihe2** [comment]
A workaround, you can write your summaries to different directories, and load them as different runs.

**Comment by b78** [comment]
@danmane
Could you explain how to inject a matplotlib plot into an image_summary. I have tried out a million things but I don't manage to.
So far I've got an image as a numpy array, but then how do I add it? Here are my last thoughts:
```
for ... training loop ...:
    sess.run(train_op)
    img = create_image(...)
    sum = tf.image_summary("foo",
                           np.reshape(img, [1, img.shape[0], img.shape[1], img.shape[2]]))
    summary_writer.add_summary(sum)
    summary_writer.flush()
```

**Comment by danmane** [comment]
@b78 Sorry, I don't have a code example on hand. This would be a good question for StackOverflow, rather than issues.

I'm not prioritizing getting two summaries into one plot, and it's probably too complicated for a pull request, so I'm going to close this issue. Maybe once TensorBoard has plugin support it will make sense to revisit this issue.

**Comment by DucVuMinh** [comment]
@while2  can you explain how to share the same name between two scalars. And can you give me an example?

**Comment by lekhamohan** [comment]
@DucVuMinh 

```
    train_graph_location = '/tmp/tensorflow_logs/
    train_writer = tf.summary.FileWriter(graph_location) #Save tensorboard variables
    test_location_location = tempfile.mkdtemp()
    test_writer = tf.summary.FileWriter(location)

```
If you print both the locations, I will get
```
/tmptensorflow_logs/train    #location of train_writer
/tmp/tmp82i59t      #test_writer
```
While loading tensorboard, I do,
`tensorboard --logdir=/tmp/`

I shoudl be able to see two graphs(different colors for distinction ofcourse) plotted on the same plot

**Comment by DucVuMinh** [comment]
@lekhamohan  Thank you for your support. I got it.

**Comment by denyHell** [comment]
I am creating multiple graph in tensorboard, and would like to have some of them printing multiple scalars ( for comparison purpose). For instance, suppose I have scalar s1,s2,...,s10, and would like s1,s2,s3 to be plotted in one graph, and s4,..., s10 to be plotted in another graph. Is there a way of doing this? Thanks in advance.
